src/common/rsvc.c

```diff
@@ -83,6 +83,10 @@ rsvc_client_choose(struct rsvc_client *client, crt_endpoint_t *ep)
 	D_DEBUG(DB_MD, DF_CLI"\n", DP_CLI(client));
 	if (client->sc_leader_known && client->sc_leader_aliveness > 0) {
 		chosen = client->sc_leader_index;
+	} else if (client->sc_ranks->rl_nr > 0 &&
+		   client->sc_next >= client->sc_ranks->rl_nr) {
```

Comment:
Hmm, does this patch work? It looks like sc_next is always < client->sc_ranks->rl_nr due to the "%" operation.